lib/Controller/MessageApiController.php

```diff
@@ -0,0 +1,171 @@
+			}
+		}
+
+		if(empty($to)) {
+			return new DataResponse('Recipients cannot be empty.', Http::STATUS_BAD_REQUEST);
+		}
+
+		$attachments = $this->request->getUploadedFile('attachments');
+		$numberOfAttachments = count($attachments['name'] ?? []);
+
+		$localAttachments = [];
+		$messageAttachments = [];
+		$attachmentErrors = false;
+		while ($numberOfAttachments > 0) {
+			$numberOfAttachments--;
+			$filedata = [
+				'name' => $attachments['name'][$numberOfAttachments],
+				'type' => $attachments['type'][$numberOfAttachments],
+				'size' => $attachments['size'][$numberOfAttachments],
+				'tmp_name' => $attachments['tmp_name'][$numberOfAttachments],
+			];
+			$file = new UploadedFile($filedata);
+			try {
+				$localAttachment = $this->attachmentService->addFile($this->userId, $file);
+				$messageAttachments[] = $localAttachment;
+				$localAttachments[] = ['type' => 'local', 'id' => $localAttachment->getId()];
+			} catch (UploadException $e) {
+				$this->logger->error('Could not convert attachment to local attachment.', ['exception' => $e]);
+				$attachmentErrors = true;
+			}
+		}
+
+		if ($attachmentErrors) {
+			foreach ($localAttachments as $localAttachment) {
+				// Handle possible dangling local attachments
+				$this->attachmentService->deleteAttachment($this->userId, $localAttachment['id']);
+			}
+			return new DataResponse('Could not convert attachment(s) to local attachment(s). Please check the logs.', Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+
+		$message = new LocalMessage();
+		$message->setType(LocalMessage::TYPE_OUTGOING);
+		$message->setAccountId($accountId);
+		$message->setSubject($subject);
+		$message->setBody($body);
+		$message->setEditorBody($body);
+		$message->setHtml($isHtml);
+		$message->setSendAt($this->time->getTime());
+		$message->setType(LocalMessage::TYPE_OUTGOING);
+
+		if(isset($alias)) {
+			$message->setAliasId($alias->getId());
+		}
+		if(!empty($references)) {
+			$message->setInReplyToMessageId($references[0]); // needs DB changes I guess
+		}
+		if(!empty($attachments)) {
+			$message->setAttachments($messageAttachments);
+		}
+
+		$recipients = array_merge($to, $cc, $bcc);
+		foreach($recipients	as $recipient) {
+			if(!is_array($recipient)) {
+				return new DataResponse('Recipient address must be an array.', Http::STATUS_BAD_REQUEST);
+			}
+
+			if(!isset($recipient['email'])) {
+				return new DataResponse('Recipient address must contain an email address.', Http::STATUS_BAD_REQUEST);
+			}
+		}
+		$localMessage = $this->outboxService->saveMessage($mailAccount, $message, $to, $cc, $bcc, $localAttachments);
+		try {
+			$localMessage = $this->outboxService->sendMessage($localMessage, $mailAccount);
+		} catch (ServiceException $e) {
+			$this->logger->error('Processing error: could not send message', ['exception' => $e]);
+			return new DataResponse('Processing error: could not send message. Please check the logs', Http::STATUS_BAD_REQUEST);
+		} catch (Exception | \Throwable $e) {
```

Comment:
```suggestion
		} catch (\Throwable $e) {
```

throwable is exception's parent